tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_friendship.c

```diff
@@ -616,6 +656,65 @@ static void test_lpn_overflow(void)
 		     msg.seq, msg.ctx.addr);
 	}
 
+	LOG_INF("Testing overflow with unsegmented message preempting segmented one...");
+
+	ASSERT_OK_MSG(bt_mesh_lpn_poll(), "Poll failed");
+
+	/* Last seq from the previous step. */
+	exp_seq = CONFIG_BT_MESH_FRIEND_QUEUE_SIZE + 1;
+
+	exp_seq += 2; /* Skipping the first message in Friend Queue. */
+	ASSERT_OK_MSG(bt_mesh_test_recv_msg(&msg, K_SECONDS(2)),
+		      "Receive first unseg msg failed");
+	ASSERT_EQUAL(5, msg.len);
+	ASSERT_EQUAL(cfg->addr, msg.ctx.recv_dst);
+	ASSERT_EQUAL(exp_seq, msg.seq);
```

Comment:
Nitpick: This check is repeated 4-5 times, so it could be converted to a helper function. Up to you